resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,307 @@
+    }
+
+    [TimeZone] Get() {
+        $currentState = [TimeZone]::new()
+        $currentState.SetTimeZoneAutomatically = [TimeZone]::GetTimeZoneAutoUpdateStatus()
+        $currentState.AdjustForDaylightSaving = [TimeZone]::GetDayLightSavingStatus()
+
+        return $currentState
+    }
+
+    [void] Set() {
+        if ($this.Test()) {
+            return
+        }
+
+        $currentState = $this.Get()
+
+        if ($currentState.TimeZone -ne $this.TimeZone) {
+            Set-TimeZone -Id (Get-ValidTimeZone -TimeZone $this.TimeZone)
+        }
+
+        if ($currentState.SetTimeZoneAutomatically -ne $this.SetTimeZoneAutomatically) {
+            $desiredState = $this.SetTimeAutomatically ? [TimeZone]::NtpEnabled : [TimeZone]::NtpDisabled
+
+            Set-ItemProperty -Path $global:tzAutoUpdatePath -Name ([TimeZone]::SetTimeZoneAutomaticallyProperty) -Value $desiredState
+        }
+
+        if ($currentState.AdjustForDaylightSaving -ne $this.AdjustForDaylightSaving) {
+            $desiredState = $this.AdjustForDaylightSaving ? 0 : 1
+
+            Set-ItemProperty -Path $global:timeZoneInformationPath -Name ([TimeZone]::AdjustForDaylightSavingProperty) -Value $desiredState
+        }
+    }
+
+    [bool] Test() {
+        $currentState = $this.Get()
+
+        if (($null -ne $this.TimeZone) -and ($this.TimeZone -ne $currentState.TimeZone)) {
+            return $false
+        }
+
+        if (($null -ne $this.SetTimeZoneAutomatically) -and ($this.SetTimeZoneAutomatically -ne $currentState.SetTimeZoneAutomatically)) {
+            return $false
+        }
+
+        if (($null -ne $this.AdjustForDaylightSaving) -and ($this.AdjustForDaylightSaving -ne $currentState.AdjustForDaylightSaving)) {
+            return $false
+        }
+
+        return $true
+    }
+
+    #region Time helper functions
+    static [bool] GetTimeZoneAutoUpdateStatus() {
+        # key should actually always be present, but we'll check anyway
+        $keyValue = TryGetRegistryValue -Key $global:tzAutoUpdatePath -Property ([TimeZone]::SetTimeZoneAutomaticallyProperty)
+        if ($null -eq $keyValue) {
+            return $true
+        } else {
+            return ($keyValue -eq 1)
+        }
+    }
+
+    static [bool] GetDayLightSavingStatus() {
+        # key should actually always be present, but we'll check anyway
+        $keyValue = TryGetRegistryValue -Key $global:timeZoneInformationPath -Property ([TimeZone]::SetTimeZoneAutomaticallyProperty)
+        if ($null -eq $keyValue) {
+            return $true
+        } else {
+            return ($keyValue -eq 0)
+        }
+    }
+
+    # helper function for Pester tests
+    [hashtable] ToHashTable() {
+        $parameters = @{}
+        foreach ($property in $this.PSObject.Properties) {
+            if (-not ([string]::IsNullOrEmpty($property.Value))) {
+                $parameters[$property.Name] = $property.Value
+            }
+        }
+
+        return $parameters
```

Comment:
I would remove this since it's not being used by the tests. I'd rather be explicit with the args we use in tests anyways since that's how they will be invoked.

Reply:
The reason for adding this, is that it is _actually_ used in the test. It can be a bit annoying for users when they run them, that the system gets modified. I'm using this as helper function to get the _current_ state before running the test and after they've run, put back the original state. If you've other recommendations, please let me know.